tools-local/init-build.proj

```diff
@@ -181,4 +173,14 @@
     <Message Importance="High" Text="[$([System.DateTime]::Now.ToString('HH:mm:ss.ff'))] Done getting source code for repos in Version.Details.xml." />
   </Target>
 
+  <Target Name="WriteSourceLinkMetadata" Condition="'$(OfflineBuild)' != 'true'" DependsOnTargets="BuildDarc;FetchSources">
+    <Message Importance="High" Text="[$([System.DateTime]::Now.ToString('HH:mm:ss.ff'))] Writing out SourceLink metadata." />
+    <WriteSourceRepoProperties
+       VersionDetailsPath="$(ProjectDir)eng/Version.Details.xml"
+       GitDirPath="$(ClonedSubmoduleGitRootDirectory)"
+       SourceDirPath="$(ClonedSubmoduleDirectory)"
```

Comment:
By convention (IMO) `Path` is not needed, `Dir` is a conventional suffix meaning a path to a directory. (Personally I like `Dir` for dirs and `File` for files, very unambiguous.)

Reply:
Looking at:
- VersionDetailsFile
- GitDirParentDir (ick)
- SourceDirParentDir (also ick)
- SourceBuildMetadataDir

Any opinion on those middle two?  It's the parent directory where we put all the .gitdirs, and the parent directory of all the source directories.

Reply:
Maybe just use the property names?

```
ClonedSubmoduleGitRootDirectory="$(ClonedSubmoduleGitRootDirectory)"
ClonedSubmoduleDirectory="$(ClonedSubmoduleDirectory)"
```

I think this is fairly normal when the task is called in a very specific scenario.